Content.Server/GameObjects/Components/HandheldRadioComponent.cs

```diff
@@ -79,12 +78,38 @@ public bool UseEntity(UseEntityEventArgs eventArgs)
 
         public void HeardSpeech(string speech, IEntity source)
         {
-            PassOnMessage(speech);
+            if (RadioOn)
+            {
+                Broadcast(speech, source);
+            }
         }
 
         public int GetListenRange()
         {
             return _listenRange;
         }
+
+        public void Receiver(string message, int channel, IEntity speaker)
+        {
+            if(RadioOn)
+            {
+                Speaker(message);
+            }
+        }
+
+        public void Broadcast(string message, IEntity speaker)
```

Comment:
Just food for thought but you could make "HeardSpeech" a bool and then move more of these methods into the system so they're in the one place.